packages/hint-create-element-svg/src/hint.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * @fileoverview Inform users that they need to use createElementNS to create SVG elements instead of createElement
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, ElementFound, IAsyncHTMLElement, ProblemLocation } from 'hint/dist/src/lib/types';
+import { debug as d } from 'hint/dist/src/lib/utils/debug';
+import { ScriptEvents, ScriptParse } from '@hint/parser-javascript';
+import { findProblemLocation } from 'hint/dist/src/lib/utils/location-helpers';
+
+import { Linter } from 'eslint';
+import * as ESTree from 'estree';
+
+import meta from './meta';
+import svgElements from './svgElements';
+
+const debug: debug.IDebugger = d(__filename);
+
+export default class CreateElementSvgHint implements IHint {
+
+    public static readonly meta = meta;
+
+    /** Finds the approximative location in the page's HTML for a match in an element. */
+    public findProblemLocation(element: IAsyncHTMLElement, content?: string): Promise<ProblemLocation> {
+        return findProblemLocation(element, { column: 0, line: 0 }, content);
+    }
+
+    public constructor(context: HintContext<ScriptEvents>) {
+        const linter = new Linter();
+
+        /*
+         * Check to see if any call expression in the AST has a callee with
+         * a 'createElement' property. If yes, check if that call expression's
+         * first argument is an SVG element. In that case, report an error as
+         * the correct identifier property for creating SVG elements is
+         * 'createElementNS'
+         */
+        linter.defineRule('svg-create', {
+            create(eslintContext) {
+                return {
+                    CallExpression(node) {
+                        const exp = node as ESTree.CallExpression;
```

Comment:
@molant and @sarvaje I'd be curious to hear your thoughts as well, but if we rely a bit more on duck-typing we can help TypeScript figure out the types automatically instead of requiring casts (it even removes the need to import `ESTree`). 

Also I'm a fan of using conditional returns at the beginning of functions to clearly call out separate checks and reduce nesting:
```javascript
CallExpression(node) {
    if (!('callee' in node && 'property' in node.callee && 'name' in node.callee.property)) {
        return;
    }

    if (node.callee.property.name !== 'createElement') {
        return;
    }

    const arg = node.arguments[0];

    if (arg && 'value' in arg && typeof arg.value === 'string' && svgElements.has(arg.value.toLowerCase())) {
        eslintContext.report({
            messageId: 'avoidElement',
            node: node.callee.property
        });
    }
}
```